### DISCUSSION TITLE
FreeTube v0.15.0 Release Candidate

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#1773] [Announcements]
Hey everyone!

For the v0.15.0 release, we're going to be trying something a little bit different. Today, we are "releasing" a test build of the upcoming release, commonly referred to as a Release Candidate. These builds are a near final revision of what we plan to release for v0.15.0 and are close if not already stable. A Release Candidate build means that we are ready to start the testing process and are close to a full release.

So that's where everyone else comes in. We have officially started our RC builds for v0.15.0 and they are now ready for testing. If anyone in the community would like to help, then they're more than welcome to do so. This thread will be for discussing these new builds as well as discussing any issues that may occur with them.

# Notes on Testing

- When you install a RC build, **this will overwrite your current version of FreeTube**. Please backup your subscriptions and other data before installing. While you will likely not have any data loss, it is best to do this anyways just in case. In the future, we will have RC builds that can be installed alongside a stable release, however for this round it will be installed on top of your current stable installation.
- When testing, please focus on the changes mentioned in our patch notes (listed below). These are the main points that we are looking at for feedback. If there's a very important issue that needs addressed before the next release that isn't mentioned, then feel free to ask about it. If your most wanted feature was not included for this release, then please do not use this thread as a method of asking for it to be included.
- Please use this thread for reporting issues if you have found something wrong with the RC builds. **Do not create an issue in our issue board**.

# How To Install

Installation for RC builds are similar to nightly builds. If you've downloaded a nightly build before, then this will be no different. **You will need to be signed into GitHub in order to access these builds**. We would love to disable this however we are unable to.

- Go to our GitHub Actions tab. [This](https://github.com/FreeTubeApp/FreeTube/actions/workflows/build.yml?query=branch%3Av0.15.0-RC) link conveniently filters our workflows to only show builds for this RC.
- Click on the first build in the list that shows a green check mark next to it.
- Scroll down on the new page to see a list of available builds. Click on the one that matches your operating system of choice.
- This next step will vary on the OS that you're using. Install the downloaded file the same way you normally would for a full release (**Again, this will overwrite your current stable installation of FreeTube**)

At this point, when you open up FreeTube you will now be using the RC build. You can confirm this by checking out the About tab and making sure that "RC" is included in the version number that's displayed.

# Patch Notes

<details>
<summary>View the patch notes for this release</summary>

## Features
- Rework of the settings page into collapsible sections
- New Dracula theme and Dracula color options
- History forward/back arrows are now grayed out when appropriate
- Caption options are now sorted based on user language preferences
- Setting to disable or ask before opening external links
- Visual improvements to Trending and channel pages
- Adds button to the search bar that clears the search box
- The search filter popup now disappears on page change

## Fixes
- Fixed a bug where opening a new video with the middle mouse button would result in a blank screen
- Fixed a bug where media control buttons were not working
- Fixed video thumbnails to be the proper width
- Comments are now formatted properly
- Fixed visual issues with dropdown & search selections on non-default themes
- Fixed a bug where menus on the mediaplayer could not be navigated while "Scroll Volume Over Video Player" was active
- Replaced links to FreeTube's deprecated GitHub wiki with https://docs.freetubeapp.io/ links
- Added proper notification messages when copying links to a playlist
- Minor label fixes
- Fixed a bug where watched videos would occasionally not be added to the watch history

## Changes
- Theater mode can now be toggled with the `T` key
- Downgraded Electron version from 14.0 to 13.5.1 
  - Commmentary: *In FreeTube v0.14.0, one of our last-minute changes was upgrading the Electron version. In doing so, we introduced a litany of Electron bugs that degraded the user experience. As a result, we will keep FreeTube's Electron version at v13.5.1 until these issues are thoroughly addressed.*
- Switched from NPM to Yarn 1 for dependency management
- Updated the README with links to Weblate & the FreeTube documentation
- Videos displayed in grid form now scale proportionally based on window size
</details>

**Comment by PrestonN** [comment]
I've created an update to the RC builds. This should fix the cert expiration issue that has recently come up. Please re-download the builds if you've done so in order to see the fix.

**Comment by senntore** [comment]
Global playback controls(in Pop OS 21.04) doesn't seem to work in v0.15.0-RC-1239 Beta, It shows unknown artist & unknown title in gnome's notification center. Tested with both .deb & appimage. You might want to open this https://github.com/FreeTubeApp/FreeTube/issues/1664 again or Should I create new issue for this version?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> Please use this thread for reporting issues if you have found something wrong with the RC builds. Do not create an issue in our issue board.

That wont be necessary. We'll look into it.

**Comment by PrestonN** [comment]
I've added an update to the RC builds. Running DASH streams from Invidious (Both with and without the proxy enabled) should be working again. This fix has also been included in the nightly builds. Feel free to leave feedback about this change here.

**Reply by ghost** [reply]
> I've added an update to the RC builds. Running DASH streams from Invidious (Both with and without the proxy enabled) should be working again. This fix has also been included in the nightly builds. Feel free to leave feedback about this change here.

The video is slow, it puts time to load, impossible to look properly. I use the nightly versions. 1277
My VPN = Mullvad

**Reply by PrestonN** [reply]
I'll make another thread explaining this a bit more. For now, you should be able to enable Invidious proxy or switching over to the Invidious API as a workaround (assuming you're using a nightly / RC build). In my testing, the throttling doesn't occur when doing this.

**Reply by ghost** [reply]
> I'll make another thread explaining this a bit more. For now, you should be able to enable Invidious proxy or switching over to the Invidious API as a workaround (assuming you're using a nightly / RC build). In my testing, the throttling doesn't occur when doing this.

Okay everything works very well with this solution, I completely disabled the local api :) Thanks
